server/lib/adapters/mysql.js

```diff
@@ -74,13 +75,13 @@ export function runQuery(client, query, options = {}) {
       }
 
       // Run the query.
-      return client.query(params, (queryError, rows) => {
+      return client.query(params, (queryError, rows, fieldPackets) => {
         client.end();
```

Comment:
@mkawalek can you look through the sync-agent and the adapters to see how close the client connection and make sure:
1. we cover all possible scenarios and possible errors (in connection, during query etc.)
2. we don't hit this `double end` error anymore